lib/ash_oban.ex

```diff
@@ -272,6 +274,14 @@ defmodule AshOban do
         doc:
           "An update action to call after the last attempt has failed. See the getting started guide for more."
       ],
+      fail_oban_job?: [
```

Comment:
Let's call this: `on_error_fails_oban_job?` and describe it as "Determines if the oban job will be failed on the last attempt when there is an `on_error` handler that is called."